### ISSUE TITLE
Split connect function props and options

### ISSUE THREAD

**Issue by timbset** [reimagined/resolve#596]
Now:
```
connectReadModel((state, props) => ({
  data: {
    item: state.readModels.items.item
  },
  readModelName: 'items',
  resolverName: 'item',
  variables: {
    id: props.itemId
  }
}));
```

It is better to make it like this:
```
connectReadModel({
  options: props => ({
    readModelName: 'items',
    resolverName: 'item',
    variables: {
      id: props.itemId
    }
  }),
  props: state => ({
    data: {
      item: state.readModels.items.item
    }
  })
})
```

If I do not need to use variables from props, I want to have possibilty to specify options as an object instead of a function. Also it is unnecessary to write `state.readModels.items.item` because I always get data here from `readModels` because I call `connectReadModel`. And also I already wrote the `readMdoelName` and `resolverName`. Why need I repeat twice? I expect to see only resolver data result, not all state structure.

**Comment by IhostVlad** [comment]
> Why need I repeat twice? I expect to see only resolver data result, not all state structure.
Maybe it is subject to rework read/view model connectors lifecycle, but now connectors just are providing automatic load and store read model data into redux store, depends on actual connected components list on SPA page - no more. Thus, component should explicitly find appropriate state part in redux and map it onto own props.

Best read/view model connectors interface is still subject to research. Most likely, whole lifecycle will be changed from post-loading to pre-loading read/view model state by react-router action or similar one. In that case whole suggestion is not applicable. 
Maybe actual implementation is not ideal, but it provides wide abilities for customisation in customer web-application, since connectors work transparently via redux store.

**Comment by IhostVlad** [comment]
It will be implemented in new `resolve-redux` implementation with prefetch adapters

**Comment by TatyanaRyzh** [comment]
Implemented in [0.13.0](https://github.com/reimagined/resolve/releases/tag/V0.13.0)
